### PR TITLE
Fixing several doc issues.

### LINKED ISSUE #486: Sample code uses a TVF that is never created (i.e. this sample code is not runnable)
Originally from https://github.com/aspnet/Release/issues/190#issuecomment-316776176 and #424.

### LINKED ISSUE #756: Please include SQL samples
Please include in this doc some generated SQL statements, or TSQL models corresponding.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d02e6748-42e0-000a-4945-42cd05a0b875
* Version Independent ID: e3effbd0-07f2-423d-4032-1722e75adb37
* Content: [Owned Entity Types - EF Core](https://docs.microsoft.com/en-us/ef/core/modeling/owned-entities)
* Content Source: [entity-framework/core/modeling/owned-entities.md](https://github.com/aspnet/EntityFramework.Docs/blob/master/entity-framework/core/modeling/owned-entities.md)
* Product: **entity-framework**
* GitHub Login: @julielerman
* Microsoft Alias: **divega**

### LINKED ISSUE #1311: Fix documentation for query filters with navigation properties
According to this GitHub issue (https://github.com/aspnet/EntityFrameworkCore/issues/8881), "Filters cannot contain references to navigation properties" limitation is not there anymore. This conflicts with the documentation.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b7c9bd3e-4b99-9acf-48d8-e47bc53eb54e
* Version Independent ID: 220e0a6c-d227-956a-d27a-7c88804f770c
* Content: [Global Query Filters - EF Core](https://docs.microsoft.com/en-us/ef/core/querying/filters)
* Content Source: [entity-framework/core/querying/filters.md](https://github.com/aspnet/EntityFramework.Docs/blob/master/entity-framework/core/querying/filters.md)
* Product: **entity-framework**
* GitHub Login: @anpete
* Microsoft Alias: **divega**

### LINKED ISSUE #2307: Clarify the interaction between filtered includes and change tracking
With filtered includes, there can now be some tricky situations where users load a filtered subset of dependents, and then attempt to add a new entity which already exists in the database. Since EF isn't aware of that entity, an INSERT will be generated and will fail. With regular include, this doesn't happen because EF knows about all the dependents from change tracking, so a different exception is raised etc. It may be worth adding a bit of guidance on this.

@smitpatel also informed me that with filtered include we consider the relationship to be loaded, e.g. for lazy-loading purposes. This should probably also be mentioned.

/cc @maumar @ajcvickers

### LINKED ISSUE #2511: Show how to do a left outer join with multiple conditions on the join
[Enter feedback here]

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: fe1acf84-126c-f01d-b39a-ab077819351f
* Version Independent ID: 5d3bdd6d-ad0e-e306-5aef-0d5adea892b6
* Content: [Complex Query Operators - EF Core](https://docs.microsoft.com/en-us/ef/core/querying/complex-query-operators)
* Content Source: [entity-framework/core/querying/complex-query-operators.md](https://github.com/dotnet/EntityFramework.Docs/blob/master/entity-framework/core/querying/complex-query-operators.md)
* Product: **entity-framework**
* Technology: **entity-framework-core**
* GitHub Login: @smitpatel
* Microsoft Alias: **avickers**

### LINKED ISSUE #2731: Breaking change in EFCore 5.0
After switching from EF3 to EF5 I get an System.InvalidOperationException on a method that previously worked:

System.InvalidOperationException: "Not enough information to uniquely identify outer element in correlated collection scenario. This can happen when trying to correlate on keyless entity or when using 'Distinct' or 'GroupBy' operations without projecting all of the key columns."

I'm afraid it is a bit complicated to reproduce since it is part of a very huge project, but I'll do my very best and can try to create a sample project, if the misbehavior is unknown and not reproducable:

I'm having a Queryable of **Offers** that gets authorized by a generic method:

```
public IQueryable<Guid> GetAuthorization(Guid paramUserId, IEnumerable<string> paramHierarchys = null, IEnumerable<string> paramPrivileges = null)
            => UserHierarchys
                .Where(h => h.Id == paramUserId && (paramHierarchys == null || paramHierarchys.Contains(h.Hierarchy)) && (paramPrivileges == null || paramPrivileges.Contains(h.Privilege)))
                .Select(h => h.CanAccessId)
                .Distinct()
            ;
```

where UserHierarchy is a simple table:

```
public class UserHierarchy : IUserHierarchy
    {
        public string Hierarchy { get; set; }
        public Guid Id { get; set; }
        public Guid CanAccessId { get; set; }
        public string Privilege { get; set; }
    }
```

Offer inherits from CreateableBase and its UserIdCreated is joined:

```
public abstract class CreateableBase : ModelBase, ICreateable
    {
        [KeepValue]
        [IgnoreHasChanged]
        public Guid? UserIdCreated { get; set; }

        [KeepValue]
        [IgnoreHasChanged]
        [ForeignKey("UserIdCreated")]
        public virtual User UserCreated { get; set; }

        [KeepValue]
        [IgnoreHasChanged]
        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public DateTime DateCreated { get; set; } = DateTime.UtcNow;
    }

public IQueryable<T> Authorize<T>(IQueryable<T> paramQueryable, Guid paramUserId, string paramHierarchy = null, string paramPrivilege = null) where T : ICreateable
            => Authorize(paramQueryable, paramUserId, p => p.UserIdCreated.Value, paramHierarchy == null ? null : new string[] { paramHierarchy }, paramPrivilege == null ? null : new string[] { paramPrivilege });
```

If an authorized Queryable now gets some includes, e.g.

```
offers = offers
                .Include(p => p.Etikettes)
                .ToArray();
```

the exception is thrown.

The query works without the Authorize / Include, but not if **both** are present. Not all includes break the program, but I couldn't figure out what includes do and what don't. Maybe some developer here has some suggestions what to check?

And there's another, maybe correlating issue I hoped to get rid of with EFCore 5.0 but don't:

I have a base class ChangeableBase inheriting from CreateableBase:

  ```
  public abstract class ChangeableBase : CreateableBase, IChangeable
    {
        [KeepValue]
        [IgnoreHasChanged]
        public Guid? UserIdChanged { get; set; }

        [KeepValue]
        [IgnoreHasChanged]
        [ForeignKey("UserIdChanged")]
        [NotMapped]
        public virtual User UserChanged { get; set; }

        [KeepValue]
        [IgnoreHasChanged]
        public DateTime? DateChanged { get; set; }
    }
```

I **need** to set UserChanged to not mapped, since the AutoMapper seams to get confused with relations of the same type pointing to different fields (UserCreated from BaseClass and UserChanged).

Thank you very much for your support!

EF Core version: 5.0.0 rc.1.20451.13
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET Core 3.1

### LINKED ISSUE #2823: EF Core 5 RC1 breaks query that works in EF Core 3.x
When running the repro code below in EF Core 5 RC1 (v5.0.0-rc.1.20451.13), it produces the next error message:

```
System.InvalidOperationException: The query contains a final projection 'a => a.ArticleTags
    .Where(at => at.Tag.Value == "x")
    .OrderBy(at => at.Tag.Id)' to type 'IOrderedEnumerable<ArticleTag>'. Collections in the final projection must be an 'IEnumerable<T>' type such as 'List<T>'. Consider using 'ToList()' or some other mechanism to convert the 'IQueryable<T>' or 'IOrderedEnumerable<T>' into an 'IEnumerable<T>'.
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryableMethodNormalizingExpressionVisitor.VerifyReturnType(Expression expression, ParameterExpression lambdaParameter)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryableMethodNormalizingExpressionVisitor.VerifyReturnType(Expression expression, ParameterExpression lambdaParameter)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryableMethodNormalizingExpressionVisitor.VerifyReturnType(Expression expression, ParameterExpression lambdaParameter)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryableMethodNormalizingExpressionVisitor.VisitMethodCall(MethodCallExpression methodCallExpression)
   at System.Linq.Expressions.MethodCallExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at Microsoft.EntityFrameworkCore.Query.QueryTranslationPreprocessor.NormalizeQueryableMethod(Expression expression)
   at Microsoft.EntityFrameworkCore.Query.RelationalQueryTranslationPreprocessor.NormalizeQueryableMethod(Expression expression)
   at Microsoft.EntityFrameworkCore.Query.QueryTranslationPreprocessor.Process(Expression query)
   at Microsoft.EntityFrameworkCore.Query.RelationalQueryTranslationPreprocessor.Process(Expression query)
   at Microsoft.EntityFrameworkCore.Query.QueryCompilationContext.CreateQueryExecutor[TResult](Expression query)
   at Microsoft.EntityFrameworkCore.Storage.Database.CompileQuery[TResult](Expression query, Boolean async)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.CompileQueryCore[TResult](IDatabase database, Expression query, IModel model, Boolean async)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.<>c__DisplayClass9_0`1.<Execute>b__0()
   at Microsoft.EntityFrameworkCore.Query.Internal.CompiledQueryCache.GetOrAddQuery[TResult](Object cacheKey, Func`1 compiler)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.Execute[TResult](Expression query)
   at Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryProvider.Execute[TResult](Expression expression)
   at Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryable`1.GetEnumerator()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at EfCoreToListBug.Program.Main(String[] args) in D:\Source\Projects\EfCoreToListBug\EfCoreToListBug\Program.cs:line 31
```

When running in EF Core v3.1.8, it succeeds without error.

Repro code:

```c#
using System.Collections.Generic;
using System.Linq;
using Microsoft.EntityFrameworkCore;

namespace EfCoreToListBug
{
    class Program
    {
        static void Main(string[] args)
        {
            using (var dbContext = new AppDbContext())
            {
                dbContext.Database.EnsureDeleted();
                dbContext.Database.EnsureCreated();

                var tag = new Tag();
                var article = new Article();
                var articleTag = new ArticleTag {Article = article, Tag =  tag};

                dbContext.ArticleTags.Add(articleTag);
                dbContext.SaveChanges();

                var query =
                    dbContext.Articles.Select(a => new Article
                    {
                        ArticleTags = new HashSet<ArticleTag>(a.ArticleTags
                            .Where(at => at.Tag.Value == "x")
                            .OrderBy(at => at.Tag.Id))
                    });

                var results = query.ToList();
            }
        }
    }

    public class AppDbContext : DbContext
    {
        public DbSet<Article> Articles { get; set; }
        public DbSet<Tag> Tags { get; set; }
        public DbSet<ArticleTag> ArticleTags { get; set; }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            optionsBuilder.UseSqlite("Data Source=sample.db");
        }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<ArticleTag>().HasKey(articleTag => new
            {
                articleTag.ArticleId, 
                articleTag.TagId
            });
        }
    }

    public abstract class Entity
    {
        public int Id { get; set; }
    }

    public class Article : Entity
    {
        public string Name { get; set; }

        public ISet<ArticleTag> ArticleTags { get; set; }
    }

    public class Tag : Entity
    {
        public string Value { get; set; }
    }

    public class ArticleTag
    {
        public int ArticleId { get; set; }
        public Article Article { get; set; }
        
        public int TagId { get; set; }
        public Tag Tag { get; set; }
    }
}
```

I believe that the "or some other mechanism" in the error message should allow the `new HashSet<ArticleTag>(...)` expression, which it fails to take into account. When replacing that with `.ToHashSet()` then the error goes away. But earlier versions of EF Core 3.x were unable to understand the `.ToHashSet()` call, which is why I'm using constructor syntax.

### PR DESCRIPTION
**Pull request by maumar**
Fixes #486
Fixes #756
Fixes #1311
Fixes #2307
Fixes #2511
Fixes #2731
Fixes #2823

### CODE REVIEW

File: entity-framework/core/what-is-new/ef-core-5.0/breaking-changes.md [hunk 1]
```diff
@@ -27,6 +27,8 @@ The following API and behavior changes have the potential to break existing appl
 | [Provider-specific EF.Functions methods throw for InMemory provider](#no-client-methods)                                              | Low        |
 | [IndexBuilder.HasName is now obsolete](#index-obsolete)                                                                               | Low        |
 | [A pluarlizer is now included for scaffolding reverse engineered models](#pluralizer)                                                 | Low        |
+| [Some queries with correlated collection that also use Distinct or GroupBy are no longer supported](collection-distinct-groupby)      | Low        |
```
Comment (maumar) [inline entity-framework/core/what-is-new/ef-core-5.0/breaking-changes.md:30]: Medium?

**Comment by AndriySvyryd** [comment]
Fix build warnings

File: entity-framework/core/querying/complex-query-operators.md [hunk 2]
```diff
@@ -24,6 +24,16 @@ FROM [PersonPhoto] AS [p0]
 INNER JOIN [Person] AS [p] ON [p0].[PersonPhotoId] = [p].[PhotoId]
 ```
 
+Further, if the key selectors are anonymous types, EF Core generates a join condition to compare equality component wise.
```
Comment (AndriySvyryd) [inline entity-framework/core/querying/complex-query-operators.md:27]:
```suggestion
Further, if the key selectors are anonymous types, EF Core generates a join condition to compare equality component-wise.
```

File: entity-framework/core/what-is-new/ef-core-5.0/breaking-changes.md [hunk 3]
```diff
@@ -390,3 +392,90 @@ Using plural forms of words for collection properties and singular forms for typ
...
+
+<a name="collection-distinct-groupby"></a>
+
+### Some queries with correlated collection that also use Distinct or GroupBy are no longer supported
+
+[Tracking Issue #15873](https://github.com/dotnet/efcore/issues/15873)
+
+**Old behavior**
+
+Previously, queries involving correlated collections followed by GroupBy, as well as some queries using Distinct we allowed to execute.
```
Comment (AndriySvyryd) [inline entity-framework/core/what-is-new/ef-core-5.0/breaking-changes.md:404]: Escape all instances of GroupBy and Distinct

File: entity-framework/core/modeling/owned-entities.md [hunk 4]
```diff
@@ -31,6 +31,10 @@ If the `ShippingAddress` property is private in the `Order` type, you can use th
 
 [!code-csharp[OwnsOneString](../../../samples/core/Modeling/OwnedEntities/OwnedEntityContext.cs?name=OwnsOneString)]
 
+The model above is mapped to the following database schema:
+
+![image](_static/owned-entities-ownsone.png)
```
Comment (AndriySvyryd) [inline entity-framework/core/modeling/owned-entities.md:36]: You need to provide readable alt text describing the image instead of just "image"

**Comment by bricelam** [comment]
Did you run the images through something like [Image Optimizer](https://marketplace.visualstudio.com/items?itemName=MadsKristensen.ImageOptimizer)?

**Comment by maumar** [comment]
updated and rebased @AndriySvyryd @bricelam

**Review by bricelam — APPROVED** [review]
LGTM

**Pull request merged**